src/main/java/org/mqttbee/mqtt/handler/MqttWebSocketsChannelInitializer.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2018 The MQTT Bee project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.mqttbee.mqtt.handler;
+
+
+import io.netty.channel.ChannelPipeline;
+import io.netty.channel.socket.SocketChannel;
+import io.netty.handler.codec.http.*;
+import io.netty.handler.logging.LogLevel;
+import io.netty.handler.logging.LoggingHandler;
+import io.netty.util.internal.logging.InternalLoggerFactory;
+import io.netty.util.internal.logging.Slf4JLoggerFactory;
+import io.reactivex.SingleEmitter;
+import io.reactivex.exceptions.Exceptions;
+import org.mqttbee.annotations.NotNull;
+import org.mqttbee.api.mqtt.mqtt5.message.connect.connack.Mqtt5ConnAck;
+import org.mqttbee.mqtt.MqttClientData;
+import org.mqttbee.mqtt.handler.websocket.WebSocketClientProtocolHandlerOnHandshakeComplete;
+import org.mqttbee.mqtt.message.connect.MqttConnect;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+/**
+ * WebSockets channel initializer.
+ *
+ * @author David Katz
+ */
+public class MqttWebSocketsChannelInitializer extends MqttChannelInitializer {
+
+    MqttWebSocketsChannelInitializer(
+        @NotNull final MqttConnect connect, @NotNull final SingleEmitter<Mqtt5ConnAck> connAckEmitter,
+        @NotNull final MqttClientData clientData) {
+
+        super(connect, connAckEmitter, clientData);
+    }
+
+    @Override
+    protected void initChannel(final SocketChannel channel) {
+        init(channel);
+        final ChannelPipeline pipeline = getPipeline();
+
+        try {
+            final URI serverUri = new URI("ws://" + clientData.getServerHost() + ":" + clientData.getServerPort() + "/mqtt");
+            final WebSocketClientProtocolHandlerOnHandshakeComplete wsProtocolHandler = new WebSocketClientProtocolHandlerOnHandshakeComplete(serverUri);
+
+            wsProtocolHandler.onUserEventTriggered(this::addMqttHandlers);
+
+            InternalLoggerFactory.setDefaultFactory(Slf4JLoggerFactory.INSTANCE);
+
+            pipeline.addLast("http-codec", new HttpClientCodec());
+            pipeline.addLast("http-aggregator", new HttpObjectAggregator(165536));
```

Comment:
65536?

Reply:
Please use separator for long numbers: 65_536

Reply:
I am not sure how large this value should be. As big as Max size for an MQTT message?

Reply:
set to 268_435_455 + 2, which is max size of an MQTT message